drracket/drracket/sprof.rkt

```diff
@@ -140,42 +136,43 @@
-                       [stacks (filter-stacks (cdr pr))])
-                   (for-each (λ (stack) (hash-set! denom-ht stack #t)) stacks)
-                   (cons id stacks)))
-               gui-display-data)]
-             [denom-count (hash-count denom-ht)])
-        (let loop ([prs filtered-gui-display-data]
-                   [first? #t]
-                   [i 0])
-          (cond
-            [(null? prs) (void)]
-            [else
-             (let* ([pr (car prs)]
-                    [fn (car pr)]
-                    [count (length (cdr pr))])
-               (cond
-                 [(zero? count)
-                  (loop (cdr prs) first? i)]
-                 [else
-                  (unless first? (insert "\n"))
-                  (let ([before (last-position)])
-                    (hash-set! line-to-source i pr)
-                    (insert (format-percentage (/ count denom-count)))
-                    (insert (format " ~a" (format-fn-name fn)))
-                    (let ([after (last-position)])
-                      (when (equal? (car pr) clicked-srcloc-pr)
-                        (set! clear-old-pr (highlight-range before after "NavajoWhite")))))
-                  (loop (cdr prs) #f (+ i 1))]))]))
-        (lock #t)
-        (end-edit-sequence)
-        (update-info-editor clicked-srcloc-pr)
-        (send open-button enable (and clicked-srcloc-pr (path? (srcloc-source (cdr clicked-srcloc-pr)))))))
+      (define denom-ht (make-hasheq))
+      (define filtered-gui-display-data
+        (map (λ (pr)
+               (let ([id (car pr)]
+                     [stacks (filter-stacks (cdr pr))])
+                 (for-each (λ (stack) (hash-set! denom-ht stack #t)) stacks)
+                 (cons id stacks)))
+             gui-display-data))
+      (define denom-count (hash-count denom-ht))
+      (let loop ([prs filtered-gui-display-data]
+                 [first? #t]
+                 [i 0])
+        (cond
+          [(null? prs) (void)]
+          [else
+           (let* ([pr (car prs)]
+                  [fn (car pr)]
+                  [count (length (cdr pr))])
+             (cond
+               [(zero? count) (loop (cdr prs) first? i)]
+               [else
+                (unless first?
+                  (insert "\n"))
+                (let ([before (last-position)])
+                  (hash-set! line-to-source i pr)
+                  (insert (format-percentage (/ count denom-count)))
+                  (insert (format " ~a" (format-fn-name fn)))
+                  (let ([after (last-position)])
+                    (when (equal? (car pr) clicked-srcloc-pr)
+                      (set! clear-old-pr (highlight-range before after "NavajoWhite")))))
+                (loop (cdr prs) #f (+ i 1))]))]))
+      (lock #t)
+      (end-edit-sequence)
+      (update-info-editor clicked-srcloc-pr)
+      (send open-button
+            enable
+            (and clicked-srcloc-pr (path? (srcloc-source (cdr clicked-srcloc-pr))))))
```

Comment:
Here's another place where `send` could probably benefit from that special treatment we discussed earlier.

Reply:
Fixed in https://github.com/sorawee/fmt/pull/86